File: Golang/Sum of All Odd Length Subarrays.go
func sumOddLengthSubarrays(arr []int) int {
    //instead of calculating all subarray sum
    //we can think about how many times for each num in arr is calculated in the subarray
    //for each nums[i], to left, there are 0, 1, i elements to the left choices, this is (i+1) choices
    //to the right, there are 0, 1, n-1-i-1+1(n-i-1), total (n-i) choices
    //and ((i+1) * (n-i) + 1) / 2 choices are of odd length
    res := 0
    n := len(arr)
    for i, num := range arr {
        cnt := ((i+1)*(n-i) + 1) / 2
        res += num * cnt
    }
    return res
}

File: Python3/Sum of All Odd Length Subarrays.py
#1 Brute-Force
class Solution:
    def sumOddLengthSubarrays(self, arr: List[int]) -> int:
        n = len(arr)
        res = 0
        for l in range(1, n + 1, 2):
            for i in range(n - l + 1):
                res += sum(arr[i:i + l])
        return res

#2 Math
class Solution:
    def sumOddLengthSubarrays(self, arr: List[int]) -> int:
        res, n = 0, len(arr)
        for i, a in enumerate(arr):
            res += ((i + 1) * (n - i) + 1) // 2 * a
        return res